apps/microtvm/zephyr/template_project/src/mlperftiny/platform.cc

```diff
@@ -162,3 +173,62 @@ int8_t QuantizeFloatToInt8(float value, float scale, int zero_point) {
   }
   return (int8_t)(result);
 }
+
+// UART interrupt callback.
+void uart_irq_cb(const struct device* dev, void* user_data) {
+  while (uart_irq_update(dev) && uart_irq_is_pending(dev)) {
+    struct ring_buf* rbuf = (struct ring_buf*)user_data;
+    if (uart_irq_rx_ready(dev) != 0) {
+      for (;;) {
+        // Read a small chunk of data from the UART.
+        int bytes_read = uart_fifo_read(dev, uart_data, sizeof(uart_data));
+        if (bytes_read < 0) {
+          TVMPlatformAbort((tvm_crt_error_t)(0xbeef1));
+        } else if (bytes_read == 0) {
+          break;
+        }
+        // Write it into the ring buffer.
+        int bytes_written = ring_buf_put(rbuf, uart_data, bytes_read);
+        if (bytes_read != bytes_written) {
+          TVMPlatformAbort((tvm_crt_error_t)(0xbeef2));
+        }
+      }
+    }
+  }
+}
+
+// Initialize the UART receiver.
+void uart_rx_init(struct ring_buf* rbuf, const struct device* dev) {
```

Comment:
is uart_rx_init() called by anyone here?

Reply:
and if not, is `uart_irq_cb()`

Reply:
removed both